python/TestHarness/testers/Tester.py

```diff
@@ -635,6 +636,10 @@ def checkRunnableBase(self, options):
             if not os.environ.get(var):
                 reasons['env_vars'] = 'ENV VAR NOT SET'
 
+        for var in self.specs['env_vars_not_set']:
+            if os.environ.get(var):
+                reasons['env_vars'] = 'ENV VAR SET'
```

Comment:
Do we output the "problematic" variable that was set?

Reply:
This isn't trivial to change. All this caveat logic is checked before the test is run to know whether we should attempt to run the test. There is no output created or saved so we'd have to refactor the test harness and do something quite different here. I would suggest we don't worry about it because we are officially using this in one place now and only for a single variable that can be inspected from the 'spec file' if it is skipped.